### ISSUE TITLE
Unable to load file '......gif' for this particular gif image

### ISSUE THREAD

**Issue by haxpor** [DCurrent/openbor#141]
Please complete the following template to help us investigate your issue. Developer time is very limited - incomplete issue reports may be ignored and deleted.

**Note:** Legacy builds are not supported. Any fixes will be applied to the latest engine builds.

## Description
Based on OpenBOR e0e7b56e5d6947138a6577c55614607382e0fc16 , I tried to use this [gif](https://internal.abzi.co/openbor/Zombie01Walk001-3.gif) and made sure that its first color in colormap is color to be used for transparent color in OpenBOR. I did a simple test against [Double Dragon Reloaded](https://github.com/DCurrent/Double-Dragon-Reloaded-DC) duplicating a certain character's script then swap in such image as a palette and frame for a particular animation.

After all, it reports "Unable to load file '....gif'.

## Debugging

Up on trying to find a root cause of problem. The error thrown out from `loadimg.c` at this lines after it entered into the case  https://github.com/DCurrent/openbor/blob/dc6a1516c7cce7e5472da4df32759df971a7ddfc/engine/source/gamelib/loadimg.c#L989-L992.

I guess it might be something about gif image's header or some sort, if I use different image from other mods, it works just fine. So I'm curious of why this particular one doesn't work.

I also used Imagemagick CLI tools `idenfity` to dump out information about such gif image to inspect in case there's something wrong, in general, it looks fine. You can see such dump log [here](https://gist.github.com/haxpor/6616f67a9059111e641492cb903c5f61) comparing to dump log of working williams character of Double Dragon Reloaded itself [here](https://gist.github.com/haxpor/56eff3e71a9bfe7742dd4499160ba406). Checked first position of colormap is target transparent color I'm going to use which is RGB => 0, 253, 255.

### Reproduce
Provide step by step instructions to reproduce the behavior. Include any text files, scripts, etc., along with the engine log. In some cases, we may need a download link to your entire module:
1. Download Double Dragon Reloaded [here](https://github.com/DCurrent/Double-Dragon-Reloaded-DC) to use as a base of mod to plug in our gif image in question
2. Download [this .gif image](https://internal.abzi.co/openbor/Zombie01Walk001-3.gif) then place it at `data/chars/williams/` directory. 
3. Modify `data/chars/williams/williams.txt` by setting its `palette` and walk animation (for all frames) to refer to such image. So when the character walks, we will see the sprite.
4. Run OpenBOR to play this mod (provided that I compiled and based off OpenBOR at e0e7b56e5d6947138a6577c55614607382e0fc16 commit)

**Note**: Actually this can be tested with any module you have, just plug in and make sure to set `palette` and frames of animation to refer to it but as first color in colormap used is specifically aligned with what Double Dragon use (0,253,255) then it's more convenient to use Double Dragon Reloaded one as base.

### Expected behavior
It should load image just fine, and work normally.

### Version
Please provide the SPECIFIC version the issue first appeared. This is very important, it is nearly impossible for us to pour through the entire code base to find singular issues without a starting point.

- Platform
`Linux haxpor-linux 4.18.0-17-generic #18~18.04.1-Ubuntu SMP Fri Mar 15 15:27:12 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
- Engine Build
e0e7b56e5d6947138a6577c55614607382e0fc16

## Additional context
No additional context or info.

**Comment by DCurrent** [comment]
This is not an OpenBOR bug. The log from Imageshack might look OK, but there is something seriously wrong with your image. I downloaded the file, and noticed that both of my image editors (Photoshop and GIMP) read it as an RGB color - by definition that is not a .gif and will never load in OpenBOR.

I gave the image a color table, re-saved, and it loaded without any problems. 
![Zombie01Walk001-3](https://user-images.githubusercontent.com/5900142/57178599-ffcb8100-6e40-11e9-89f8-1e5d727698c1.gif)

**Comment by haxpor** [comment]
That's weird. After testing more, I found out that very likely due to setting of `palette` because each frames of animation I used has different colormap, the less of the frames don't base off on a single same colormap. Might be possible that a certain frame tries to reach to out of bound or some sort of colormap (I just guess).

After I set to `palette none` it works fine, but I've read discussion some where in the forum, this incurs more memory usage.

Thank you DC. I might have to rely on `palette none` to make it work for now.

**Comment by dbaldan** [comment]
Don't use palette none for characters, it's a terrible idea and waste of resources

**Comment by haxpor** [comment]
@dbaldan thank for reminding. I've read that somewhere in the chronocrash forum too. For my case, I have assets which aren't designed to be limited by color palette from start, and each frame in a single animation can have variation of number of colors used. It seems to be tricky for me to stick to set `palette` option although I'd love to. I'm still exploring...

**Comment by msmalik681** [comment]
Something like this should be explored on the forum and should be confirmed as a bug before it reaches here.  That is my personal opinion.

**Comment by haxpor** [comment]
@msmalik681 ok, I kindly noted that; it's fine if community encourages to use forum first. I'll keep that in mind.

FYI: I found the problem, and it's actually a fault on my side. I created an automated tool to rearrange the position of color in colormap to be the first position in order to satisfy OpenBOR engine; can use in batch process. For some reason, as DC pointed out, colormap is not set. That leads me to solve the solution. Ok, OpenBOR works fine, not a bug. Thank you again for all.
